Extend Derive with an arbitrary long sequence of dependencies (#19295)
coq/coq: doc/sphinx/addendum/miscellaneous-extensions.rst

required with ``Require Coq.derive.Derive``. When the extension is loaded,
it provides the following command:

.. cmd:: Derive @open_binders in @type as @ident
         Derive @open_binders SuchThat @type As @ident

>    where :n:`@open_binders` declares a list of declarations of the form

Reviewer: ```suggestion
   where :n:`@open_binders` is a list of the form
```

>    :g:`@ident__i : @type__i` which can appear in :n:`@type`. This

Reviewer: Will you generate error messages if the open_binders have something other than `@ident : @type`?

Author: Yes, there is an error message if they don't have the expected form.

   command opens a new proof presenting the user with a goal for

>    :n:`@type` in which each name :n:`@ident__i` is bound to an
>    existential variable :g:`?x__i` (formally, there are other goals

Reviewer: ```suggestion
   :n:`@type` in which each :n:`@ident__i` is bound to an
   existential variable :n:`?x__i` (formally, there are other goals
```

Ah, these will be fresh existential variables, right, so it's misleading to specifically say :n:`?x__i`; they may be numbered differently./

Author: Not exactly, the existential variables will have the same name as the constant. Maybe should I used `?name_i` instead?

Reviewer: I don't know.  I really don't understand the command.  Sounds like `?ident__i` is sort of right but the numbers may not match up.  Maybe just be less specific, e.g. just say "bound to a new existential variable beginning with :n:`@ident__i` (formally ..." and don't give any `?XXX` in the text.

   standing for the existential variables but they are shelved, as
   described in :tacn:`shelve`).

>    When the proof ends as many :term:`constants <constant>` as the
>    number of :g:`@ident__i` plus one are defined:

Reviewer: ```suggestion   
   When the proof is complete, Coq defines :term:`constants <constant>` for each :n:`@ident__i` and for :n:`@ident`:
```

Would you clarify what you mean by defining a constant?  Is it an identifier tied to the proof or is visible outside the proof, such as for the `Inductive` command?
